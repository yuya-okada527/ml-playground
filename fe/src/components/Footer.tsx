import {
  createStyles,
  List,
  ListItem,
  makeStyles,
  Theme,
} from "@material-ui/core";
import React from "react";

const MAIL_ADDRESS = "yuya.okada527@gmail.com";
const GITHUB_URL = "https://github.com/yuya-okada527/ml-playground";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      width: "100%",
      height: "65px",
      borderTop: "1px solid black",
      marginTop: theme.spacing(2),
    },
    footerList: {
      paddingTop: theme.spacing(0.5),
    },
    footerListItem: {
      paddingTop: theme.spacing(0.5),
      paddingBottom: theme.spacing(0),
    },
  })
);

/**
 * フッターを描画する
 */
const Footer: React.FC = () => {
  const classes = useStyles();
  return (
    <footer className={classes.footer}>
      <List className={classes.footerList}>
        <ListItem className={classes.footerListItem}>
          Contact: {MAIL_ADDRESS}
        </ListItem>
        <ListItem className={classes.footerListItem}>
          Github: {GITHUB_URL}
        </ListItem>
      </List>
    </footer>
  );
};

export default Footer;
