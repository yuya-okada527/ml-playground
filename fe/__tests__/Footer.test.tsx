import React from "react";
import { shallow } from "enzyme";
import Footer from "../src/components/Footer";

describe("Footer", () => {
  it("Snapshot Test", () => {
    const footer = shallow(<Footer />);
    expect(footer).toMatchSnapshot();
  });
});
