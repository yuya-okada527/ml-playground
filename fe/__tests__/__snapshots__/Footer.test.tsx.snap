// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Test 1`] = `
<footer
  className="makeStyles-footer-1"
>
  <WithStyles(ForwardRef(List))
    className="makeStyles-footerList-2"
  >
    <WithStyles(ForwardRef(ListItem))
      className="makeStyles-footerListItem-3"
    >
      Contact: 
      yuya.okada527@gmail.com
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      className="makeStyles-footerListItem-3"
    >
      Github: 
      https://github.com/yuya-okada527/ml-playground
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</footer>
`;
